src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -846,17 +846,15 @@ int DefaultPolicy::DetermineCallsiteNativeSizeEstimate(CORINFO_METHOD_INFO* meth
 {
     int callsiteSize = 55; // Direct call take 5 native bytes; indirect call takes 6 native bytes.
 
-    bool hasThis = methInfo->args.hasThis();
-
-    if (hasThis)
+    if (methInfo->args.hasImplicitThis())
     {
         callsiteSize += 30; // "mov" or "lea"
     }
 
     CORINFO_ARG_LIST_HANDLE argLst = methInfo->args.args;
     COMP_HANDLE             comp   = m_RootCompiler->info.compCompHnd;
 
-    for (unsigned i = (hasThis ? 1 : 0); i < methInfo->args.totalILArgs(); i++, argLst = comp->getArgNext(argLst))
+    for (unsigned i = 0; i < methInfo->args.numArgs; i++, argLst = comp->getArgNext(argLst))
```

Comment:
Since I worked through it, I may as well write it here:

| kind | old behavior | new behavior
|--|--|--
|no-this|ok|ok
|hasthis|this handled above, normal args ok|same
hasthis and explicitthis|this handled above, this skipped here, normal args ok|this skipped above, this handled here, normal args ok

1. Only "change" is explicitthis being handled in loop, but it's going to be `+= 30` either way
2. This is clearer.
👍